lib/cylc/task_pool.py

```diff
@@ -1037,24 +1032,17 @@ def spawn_tasks(self, items):
         return len(bad_items)
 
     def reset_task_states(self, items, status, outputs):
-        """Reset task states."""
+        """Operator-forced task status reset and output manipulation."""
         itasks, bad_items = self.filter_task_proxies(items)
         for itask in itasks:
             if status and status != itask.state.status:
                 LOG.info("resetting state to %s" % status, itask=itask)
-                if status == TASK_STATUS_READY:
-                    # Pseudo state (in this context) -
-                    # set waiting and satisified.
-                    itask.state.reset_state(TASK_STATUS_WAITING)
-                    itask.state.set_prerequisites_all_satisfied()
-                    itask.state.unset_special_outputs()
-                    itask.state.outputs.set_all_incomplete()
-                else:
-                    itask.state.reset_state(status)
-                    if status in [
-                            TASK_STATUS_FAILED, TASK_STATUS_SUBMIT_FAILED]:
-                        itask.set_event_time('finished',
-                                             get_current_time_string())
+                itask.state.reset_state(status)
+                if status in [TASK_STATUS_FAILED,
+                              TASK_STATUS_SUBMIT_FAILED]:
+                    # TODO - HUH? SUBMIT_FAILED? WHAT ABOUT SUCCEEDED?
+                    itask.set_event_time('finished',
```

Comment:
This looks suspicious, indeed. Needs some more explanation if this is correct.

Reply:
Should probably include all states in `cylc.task_state.TASK_STATUSES_FINAL`.

Reply:
(oops, looks like I forgot to come back to my own reminder there!)

Reply:
Fixed.  Turns out `set_event_time` merely updates the state summary times shown in the GUI - nothing to do with "events" as such.  So I've changed the method name, and only update the "finished" time for succeeded and failed tasks.